srcs/app/CGI/CgiHandler.cpp

```diff
@@ -1,22 +1,33 @@
 #include <CgiHandler.hpp>
 
-CgiHandler::CgiHandler(int socket, int cgi_output)
-	: socket_(socket), cgi_output_(cgi_output), cgi_complete_(false) {}
+CgiHandler::CgiHandler(FDsets *fd_sets, int socket, const CgiInfo &cgi_info)
+	: fd_sets_(fd_sets), socket_(socket), cgi_info_(cgi_info),
+	cgi_complete_(false), headers_parsing_complete_(false), status_(200) {
+	fd_sets_->addToReadSet(cgi_info_.cgi_output_fd);
+}
 
 CgiHandler::~CgiHandler() {
+	fd_sets_->removeFd(socket_);
```

Comment:
It looks very clean after you moved these operations on the fd_sets to the constructor and destructor of this class! Thanks!

Reply:
Thank you!